### ISSUE TITLE
RFE: common interface to expose host's os-release to container-like runtimes

### ISSUE THREAD

**Issue by bluca** [lxc/lxc#3426]
The following was the start of a discussion regarding the possibility of establishing a common interface for exposing the host's os-release via container runtimes:

> We find ourselves in a situation where application running
> containerized (specifically as portable services, although this issue
> is general) need to either find information about the host, or alter
> their behaviour depending on the host's flavour.
> At the same time, applications also need to find information about the
> guest's OS, so simply bind mounting /etc/os-release from the host is
> not enough. We need both.

Reference: https://lists.freedesktop.org/archives/systemd-devel/2020-April/044378.html

The desired goal is to establish a common interface to provide this information, so that applications can rely on it without having to infer what runtime manager they are spawned by.

There are many ways to implement this, the discussion for the nspawn/portabled cases gravitated toward using prefixed environment variables - the advantage being that it's the most convenient way for applications, since they can trivially have both sets of information at the same time (eg: source /etc/os-release and then VERSION_ID is the runtime's version, container_host_VERSION_ID is the host's version).

RFC patches for nspawn/portabled/flatpak:

https://github.com/systemd/systemd/pull/15891
https://github.com/flatpak/flatpak/pull/3640

Thoughts?

**Comment by brauner** [comment]
@stgraber

**Comment by brauner** [comment]
You can tell LXC to set any environment variable you want anyway already. I don't much see the point of setting this by default. That potentially seems like an information leak into the container especially considering running full distros in there.

**Comment by stgraber** [comment]
We will not be doing something out of the box, this is information leakage that may facilitate a host attack from the container.
It's certainly something you can script around by passing a bind-mount or an env variable, but not something we'd ever want to do by default.

**Comment by bluca** [comment]
> We will not be doing something out of the box, this is information leakage that may facilitate a host attack from the container.
> It's certainly something you can script around by passing a bind-mount or an env variable, but not something we'd ever want to do by default.

Hello Stéphane, thanks for taking the time to look at this.

Would it be acceptable if when doing the work we added a switch, so that it's not enabled by default? My main desire was to have a common way of accessing this, given os-release is a well-defined cross-distro interface.
I know it can be done in other ways - in fact we already do, and so do others - I counted at least 4 different ways. Would be nice to go down to one, that's all :-)
